server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -63,9 +69,21 @@ public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
 
         // determine count
         // default counts
-        BigDecimal count = isText ? new BigDecimal(DEFAULT_TEXT_COUNT) : new BigDecimal(DEFAULT_NUMBER_COUNT);
-        if (this.clusterClause.getCount() != 0) {
-            count = new BigDecimal(this.clusterClause.getCount());
+        BigDecimal count = new BigDecimal(this.clusterClause.getCount());
+        if (count.compareTo(new BigDecimal(0)) == 0) {
+            switch (fieldType) {
+                case KEYWORD:
+                case TEXT:
+                    count = new BigDecimal(DEFAULT_TEXT_COUNT);
+                    break;
+                case DATETIME:
+                    count = new BigDecimal(DEFAULT_DATETIME_COUNT);
+                    break;
+                case DECIMAL:
+                default:
+                    count = new BigDecimal(DEFAULT_NUMBER_COUNT);
+                    break;
```

Comment:
I think the default cause should just be not clustering at all?

Reply:
Something felt off, that's a good call!

Reply:
It now returns the data unclustered on the count switch. This is the first switch so it should take care of that case.

Reply:
Looks great!